### ISSUE TITLE
Keep the image file name and avoid duplicate file

### ISSUE THREAD

**Issue by Light-Studio** [thedevdojo/voyager#2509]
- Laravel Version: 5.5
- Voyager Version: 1.0
- Database Driver & Version:

### Description:
Hi, I'm just sharing a tips to keep the filename and add some random character to avoid duplicate name of your uploaded images since it's good to have a nice filename for SEO purpose. 

### Steps To Reproduce:

With this method you will obtain something like this "**filename_sdGEn.jpg**"

For that we have to make modification to the /vendor/tcg/voyager/src/Http/Controllers/Controller.php file (I know is not the best to edit directly in the vendor's file but i did not find a way to override this one)

go to the line  323

where it's mark
`$filename = Str::random(20)`

you can replace it by
`$filename = $nameonly . '_' . Str::random(5);`
(I've put 5 in **Str:random()** for keeping a short name fell free to put more random letters if you want)

now above this line you can put this : 

```ruby
$jpg = ends_with($file->getClientOriginalName(), '.jpg');
$jpeg = ends_with($file->getClientOriginalName(), '.jpeg');
$png =ends_with($file->getClientOriginalName(), '.png');
$gif =ends_with($file->getClientOriginalName(), '.gif');
                        
if($jpg == true)
                            
$nameonly= str_slug(str_before($file->getClientOriginalName(),'.jpg'), '_');
                        
elseif($jpeg == true)
                            
$nameonly= str_slug(str_before($file->getClientOriginalName(),'.jpeg'), '_');
                        
elseif($png == true)
                            
$nameonly= str_slug(str_before($file->getClientOriginalName(),'.png'), '_');
                        
elseif($gif == true)
                            
$nameonly= str_slug(str_before($file->getClientOriginalName(),'.gif'), '_');

```

I know it's a lot of line to get this small change but when you use "**$file->getClientOriginalName()**"
you get the filename with the extension so it's not possible to insert random characters between.

so basically we checking what is the file extension with the "**ends_with**"
Then depend of the extension we keep the filename without the extension with "**str_before**"
and we remove the space and special characters with **"str_slug"**

If you find a better or cleaner way to do that feel free to share it with us !

I hope this will help some of you !
